src/main/java/uk/gov/hmcts/ccd/domain/service/caseaccess/CaseAccessOperation.java

```diff
@@ -50,4 +59,31 @@ public void revokeAccess(final String jurisdictionId, final String caseReference
                                  .collect(Collectors.toList());
     }
 
+    @Transactional
+    public void updateUserAccess(CaseDetails caseDetails, CaseUser caseUser) {
+        final Set<String> validCaseRoles = caseRoleRepository.getCaseRoles(caseDetails.getCaseTypeId());
+        final Set<String> targetCaseRoles = caseUser.getCaseRoles();
+
+        targetCaseRoles.stream()
+                       .filter(role -> !validCaseRoles.contains(role))
+                       .findFirst()
+                       .ifPresent(role -> {
+                           throw new InvalidCaseRoleException(role);
+                       });
+
+        final Long caseId = new Long(caseDetails.getId());
+        final List<String> currentCaseRoles = caseUserRepository.findCaseRoles(caseId, caseUser.getUserId());
+
+        // Grant added case roles
```

Comment:
Same here we could convert this block of code to a method with a signature name reflecting the comment.

Reply:
That makes perfect sense, done